common/persistence/sql/sqlplugin/sqlite/plugin.go

```diff
@@ -123,11 +136,10 @@ func (p *plugin) createDBConnection(
 	// Maps struct names in CamelCase to snake without need for db struct tags.
 	db.MapperFunc(strcase.ToSnake)
 
-	// only suitable for in memory databases!
-	if cfg.ConnectAttributes["mode"] == "memory" {
+	if cfg.ConnectAttributes["mode"] == "memory" || cfg.ConnectAttributes["setup"] == "true" {
 		// creates temporary DB overlay in order to configure database and schemas
 		err := p.setupSQLiteDatabase(cfg, db)
-		if err != nil {
+		if err != nil && !IsTableExistsError(err) {
```

Comment:
I would like this code block to me more explicit:
1. Explicitly indicate which `mode` value are supported (`switch` instead of `if`).
2. Explicitly indicate that `setup` makes sense only for `file` mode.
3. Explicitly indicate that `isTableExisisError` is expected only in `file` mode.